vault/version_store.go

```diff
@@ -66,6 +69,30 @@ func (c *Core) storeVersionEntry(ctx context.Context, vaultVersion *VaultVersion
 	return true, nil
 }
 
+// storeLastUpgrade will store the last version of Vault that was successfully
```

Comment:
Isn't this a misnomer?  Don't we store this regardless of whether any upgrades ever happened?